spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -86,8 +93,23 @@ else if (tokens.length != 1) {
 								" [<transport>:]<name>");
 					}
 				}
-				Binder<MessageChannel> binder = binderFactory.getBinder(transport);
-				binder.bindProducer(name, channel, this.producerProperties);
+				String[] dynamicDestinations = null;
+				Properties producerProperties = null;
+				if (this.channelBindingServiceProperties != null) {
+					dynamicDestinations = this.channelBindingServiceProperties.getDynamicDestinations();
+					// TODO: need the props to return some defaults if not found
+					producerProperties = this.channelBindingServiceProperties.getProducerProperties(name);
+				}
+				boolean dynamicAllowed = dynamicDestinations == null
+						|| dynamicDestinations.length == 0
+						|| Arrays.asList(dynamicDestinations).contains(name);
+				if (dynamicAllowed) {
```

Comment:
Also, should we do this check before registering the channel? It seems like there would be little reason to do the channel registration unless we can actually bind to it.